tests/schema/tests.py

```diff
@@ -3191,3 +3191,164 @@ class Meta:
         with connection.schema_editor(atomic=True) as editor:
             editor.alter_db_table(Foo, Foo._meta.db_table, 'renamed_table')
         Foo._meta.db_table = 'renamed_table'
+
+    @isolate_apps('schema')
+    @skipUnlessDBFeature('supports_collation_on_charfield')
+    def test_collations(self):
+        if connection.vendor == 'mysql':
+            collation = connection.features.test_collations['swedish-ci']
+        else:
+            collation = list(connection.features.test_collations.values())[0]
```

Comment:
I would add instead a new entry to `test_collations` dictionaries, e.g. `non_default_column`?